app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb

```diff
@@ -211,7 +211,7 @@ def enabled_repos
 
     #api :POST, "/environments/:environment_id/consumers", N_("Register a consumer in environment")
     def consumer_create
-      host = Katello::RegistrationManager.process_registration(rhsm_params, find_content_view_environments)
+      host = Katello::RegistrationManager.process_registration(rhsm_params, content_view_environments)
```

Comment:
I think it should be `find_content_view_environments` (see L#321 below)

Reply:
Moved it back. I don't see this path being called from subman but may be some one who is curl posting to this URL directly will be calling this.

Reply:
Hmm, still the same issue:

```
[ActiveRecord::RecordInvalid] exception expected, not
Class: <ActiveModel::UnknownAttributeError>
Message: <"unknown attribute 'content_view' for Katello::ActivationKey.">
```

Reply:
To clarify, this is what it was originally before my PR:

```ruby
    def consumer_create
      host = Katello::RegistrationManager.process_registration(rhsm_params, find_content_view_environments) # note second arg here, calls find_content_view_environments method which is defined later in the same file

      host.reload

      update_host_registered_through(host, request.headers)
      render :json => Resources::Candlepin::Consumer.get(host.subscription_facet.uuid)
    end
```

My PR had:

```ruby
    def consumer_create
      content_view_environments = find_content_view_environments # content_view_environments is defined here
      if content_view_environments.any? { |cve| cve.content_view.generated_for_repository_export? }
        render json: {
          displayMessage: _("Cannot register with a content view generated for repository export"),
          errors: [_("Cannot register with a content view generated for repository export")]
        }, status: :bad_request
        return
      end

      host = Katello::RegistrationManager.process_registration(rhsm_params, content_view_environments) # second arg here now uses above definition

      host.reload

      update_host_registered_through(host, request.headers)
      render :json => Resources::Candlepin::Consumer.get(host.subscription_facet.uuid)
    end
```

And yours had:

```ruby
    def consumer_create
      host = Katello::RegistrationManager.process_registration(rhsm_params, content_view_environments) # but content_view_environments wasn't defined as in mine

      host.reload

      update_host_registered_through(host, request.headers)
      render :json => Resources::Candlepin::Consumer.get(host.subscription_facet.uuid)
    end
```

So I wasn't suggesting that you need to reinstate my version of the change, just that you never FULLY reverted it to the original

That said, it seems the current issue in the tests is occurring regardless. I'm not sure what is the cause

Reply:
> [ActiveRecord::RecordInvalid] exception expected, not
> Class: <ActiveModel::UnknownAttributeError>
> Message: <"unknown attribute 'content_view' for Katello::ActivationKey.">

can you share the stack trace for this.

Reply:
> can you share the stack trace for this.

I kicked off a re-run in CI of the failed tests to be sure:

```
 Error: Failure: test_0011_should not allow generated content view to be associated with activation key

[ActiveRecord::RecordInvalid] exception expected, not
Class: <ActiveModel::UnknownAttributeError>
Message: <"unknown attribute 'content_view' for Katello::ActivationKey.">
---Backtrace---
katello/test/models/activation_key_test.rb:114:in `block (2 levels) in <class:ActivationKeyTest>'
katello/test/models/activation_key_test.rb:113:in `block in <class:ActivationKeyTest>'
---------------
```

It looks like the test needs to try updating `convent_view_environments` instead